### ISSUE TITLE
Deterministic seeds?

### ISSUE THREAD

**Issue by JupiLogy** [cubing/mark3#1]
Hi, this is more of a question.

Wouldn't having deterministic seeds make it so that anyone can generate the scrambles for any comp? E.G. Suppose I'm going to the "UK champs 2025". What's the difference between me running mark3 with "UK champs 2025", "3x3", "round 1", etc. and the delegates doing the same thing? Doesn't this mean that I could see the scrambles before the competition?

**Comment by lgarron** [comment]
Indeed, this is the most important part of the whole system. In the current draft design everything is deterministically generated from the root seed of a competition, but the root seed itself is not. The properties I'm aiming for are:

- The seed is selected in an exchange between a WCA server and scramble generation client such that:
  - Only the client knows the resulting seed.
  - The server has a commitment from the client that can only be satisfied by the client revealing the seed (after the competition).
    - Exception: some sort of "break glass" mechanism that requires cooperation from multiple WMT members to access a seed, such as Shamir secret sharing encrypted to HSMs (e.g. Yubikeys).
  - Neither side can influence or predict any information about the seed before the exchange begins.
